USER_GUIDE.md

```diff
@@ -359,7 +360,64 @@ try {
 
 ```
 
-## ClientBuilder
+## Client Factories
+
+You can create an OpenSearch Client using any [PSR-18](https://www.php-fig.org/psr/psr-18/) compatible 
+HTTP client. Two factories are provided to reduce the boilerplate code required to create a client 
+using [Guzzle](https://docs.guzzlephp.org/en/stable/) or [Symfony](https://symfony.com/doc/current/http_client.html) 
+HTTP clients.
+
+The `GuzzleClientFactory` and `SymfonyClientFactory` classes are used to create an OpenSearch Client instance.
+
+### Guzzle Client Factory
+
+This factory creates an OpenSearch Client instance using the Guzzle HTTP client.
+
+```bash
+composer require guzzlehttp/guzzle
+```
+
+```php
+$client = (new \OpenSearch\GuzzleClientFactory())->create([
+    'base_uri' => 'https://localhost:9200',
+    'auth' => ['admin', getenv('OPENSEARCH_PASSWORD')],
+    'verify' => false,
+]);
+```
+`GuzzleClientFactory::__construct()` accepts an `int $maxRetries` as the first argument to enable retrying a request
+when a `500 Server Error` status code is received in the response, or network connection error occurs. A PSR-3 Logger
+can be passed as the second argument to log the retries.
+
+[Guzzle Request options](https://docs.guzzlephp.org/en/stable/request-options.html) can be passed to the 
+`create()` method. The `base_uri` option is *required*.
+
+### Symfony Client Factory
+
+This factory creates an OpenSearch Client instance using the Symfony HTTP client.
+
+```bash
+composer require symfony/http-client
+```
+
+```php
+$client = (new \OpenSearch\SymfonyClientFactory())->create([
+    'base_uri' => 'https://localhost:9200',
+    'auth_basic' => ['admin', getenv('OPENSEARCH_PASSWORD')],
+    'verify_peer' => false,
```

Comment:
maybe add a comment for testing purposes. in prod you have a valid or specify it

Reply:
Comment added.